enhancements/builds/multi-arch-builds.md

```diff
@@ -0,0 +1,456 @@
+push, create N+1 pods when building for N architectures.
+
+The first N pods will build an ordinary image for their respective targets and,
+if the build spec specifies an output location, push their images and update a
+platform-specific build status field in the build resource.
+
+The last pod will use functionality to be added to the openshift builder to
+build a manifest list and push it to the output location.
+
+### openshift-builder
+
+The builder binary is currently invoked by multiple names to perform multiple
+tasks in build pods.
+
+We'll extend openshift-builder's `openshift-docker-build` and
+`openshift-sti-build` commands to perform the build as they already do, with
+two exceptions: if the build spec specifies a target platform, the build will
+target that platform for the build, and the build status will be written to the
+corresponding platform-specific status field in the Build object which owns the
+build pod.
+
+We'll extend openshift-builder by adding another command it can be invoked by
+which will tell it to monitor the status of the other builder pods. If the
+builder pods all complete successfully, and the build spec specifies an output
+location, it will build and push a manifest list using the information gathered
+from the build status information. As its final act, it will attempt to untag
+the individual images from the registry.
+
+Note: untagging should be friendlier to registries which perform garbage
+collection, particularly for cases where the build fails on one or more
+architectures, and as a result a manifest list is not pushed. It should also
+reduce "noise" in the list of tags reported by registries which support listing
+the tags in a repository.
+
+### Workflow Description
+
+#### Variation 1/3: SCM integrated
+1. Application developer merges a PR to their repository's main branch.
+1. The SCM platform notifies OpenShift via a webhook.
+1. OpenShift has previously associated the webhook with triggering a BuildConfig.
+
+#### Variation 2/3: not SCM integrated
+1. Application developer manually triggers a BuildConfig.
+
+#### Variation 3/3: not SCM integrated
+1. Application developer creates a Build.
+
+#### Common
+1. The openshift-controller-manager starts N+1 build pods.
+1. Platform-specific images are pushed to the registry.
+1. The build completes, and a manifest list is pushed to the registry.
+1. Platform-specific images are untagged from the registry.
+
+#### Variation 1/3: no registry/DeploymentConfig integration
+1. Application developer retrieves a pullable reference to the manifest list from the build status object and updates their pod specs.
+1. Pods are restarted.
+
+#### Variation 2/3: registry/DeploymentConfig integration
+1. The registry notifies OpenShift of the newly pushed tag.
+1. DeploymentConfigs which reference this tag in an ImageChangeTrigger cause the image to be re-pulled and pods to be restarted on affected nodes.
```

Comment:
more of a question for @dmage @flavianmissi will the reference by tag pull the manifestlist images in this case where the tag was first pointing to a single manifest image, and after a new arch node was added, builds generates a manifestlist image?

Reply:
ImageChangeTrigger uses to the current image digest that is available for the image stream tag. If you push a manifest list, it'll be a digest of the manifest list.